challenger/lnd.go

```diff
@@ -270,7 +270,9 @@ func (l *LndChallenger) readInvoiceStream(
 
 // Stop shuts down the challenger.
 func (l *LndChallenger) Stop() {
-	l.invoicesCancel()
+	if l.strictVerify {
```

Comment:
Should we instead check if `l.invoicesCancel` is not nil? In case it wasn't created for another reason, independent of the `strictVerify` flag.